diff --git a/tools/build/src/tools/link.jam b/tools/build/src/tools/link.jam
index 944911b..8e5c523 100644
--- a/tools/build/src/tools/link.jam
+++ b/tools/build/src/tools/link.jam
@@ -63,31 +63,10 @@ rule can-symlink ( project : ps )
 if [ os.name ] = NT
 {
 
-# Test for Windows junctions (mklink /J)
+.can-junction = true ;
+
 rule can-junction ( project : ps )
 {
-    if ! $(.can-junction)
-    {
-        local root-project = [ get-root-project $(project) ] ;
-
-        local source-target = [ new file-target test-junction-source : :
-            $(project) : [ new action : common.mkdir ] ] ;
-        local target = [ new file-target test-junction : :
-            $(project) : [ new action $(source-target) : link.junction ] ] ;
-
-        if [ configure.try-build $(target) : $(ps) : "junctions supported" ]
-        {
-            .can-junction = true ;
-        }       
-        else
-        {
-            .can-junction = false ;
-        }
-    }
-    if $(.can-junction) = true
-    {
-        return true ;
-    }
 }
 
 }
