diff --git a/tools/build/src/tools/msvc.jam b/tools/build/src/tools/msvc.jam
index 2e481d4..513b7e9 100644
--- a/tools/build/src/tools/msvc.jam
+++ b/tools/build/src/tools/msvc.jam
@@ -1137,6 +1137,14 @@ local rule generate-setup-cmd ( version : command : parent : options * : cpu : g
         }
         else
         {
+            if [ MATCH "(14.4)" : $(version) ]
+            {
+                if $(.debug-configuration)
+                {
+                    ECHO "notice: [generate-setup-cmd] $(version) is 14.4" ;
+                }
+                parent = [ path.native [ path.join  $(parent) "..\\..\\..\\..\\..\\Auxiliary\\Build" ] ] ;
+            }
             if [ MATCH "(14.3)" : $(version) ]
             {
                 if $(.debug-configuration)
