diff -U5 -Nru hdf5-1.8.13/config/cmake/ConfigureChecks.cmake hdf5-1.8.13.vs2015-support/config/cmake/ConfigureChecks.cmake
--- hdf5-1.8.13/config/cmake/ConfigureChecks.cmake	2014-05-05 22:13:20.000000000 -0400
+++ hdf5-1.8.13.vs2015-support/config/cmake/ConfigureChecks.cmake	2016-08-08 15:44:55.565692986 -0400
@@ -181,11 +181,13 @@
   if (NOT UNIX AND NOT CYGWIN AND NOT MINGW)
     set (H5_HAVE_GETCONSOLESCREENBUFFERINFO 1)
   endif (NOT UNIX AND NOT CYGWIN AND NOT MINGW)
   set (H5_HAVE_FUNCTION 1)
   set (H5_GETTIMEOFDAY_GIVES_TZ 1)
-  set (H5_HAVE_TIMEZONE 1)
+  if (NOT MSVC14)
+    set (H5_HAVE_TIMEZONE 1)
+  endif ()
   set (H5_HAVE_GETTIMEOFDAY 1)
   set (H5_LONE_COLON 0)
   if (MINGW)
     set (H5_HAVE_WINSOCK2_H 1)
   endif (MINGW)
@@ -900,39 +902,38 @@
 # Check how to print a Long Long integer
 #-----------------------------------------------------------------------------
 if (NOT H5_PRINTF_LL_WIDTH OR H5_PRINTF_LL_WIDTH MATCHES "unknown")
   set (PRINT_LL_FOUND 0)
   message (STATUS "Checking for appropriate format for 64 bit long:")
-  foreach (HDF5_PRINTF_LL l64 l L q I64 ll)
-    set (CURRENT_TEST_DEFINITIONS "-DPRINTF_LL_WIDTH=${HDF5_PRINTF_LL}")
-    if (H5_SIZEOF_LONG_LONG)
-      set (CURRENT_TEST_DEFINITIONS "${CURRENT_TEST_DEFINITIONS} -DHAVE_LONG_LONG")
-    endif (H5_SIZEOF_LONG_LONG)
-    TRY_RUN (HDF5_PRINTF_LL_TEST_RUN   HDF5_PRINTF_LL_TEST_COMPILE
-        ${HDF5_BINARY_DIR}/CMake
-        ${HDF5_RESOURCES_DIR}/HDF5Tests.c
-        CMAKE_FLAGS -DCOMPILE_DEFINITIONS:STRING=${CURRENT_TEST_DEFINITIONS}
-        OUTPUT_VARIABLE OUTPUT
+  set (CURRENT_TEST_DEFINITIONS "-DPRINTF_LL_WIDTH")
+  if (H5_SIZEOF_LONG_LONG)
+    set (CURRENT_TEST_DEFINITIONS "${CURRENT_TEST_DEFINITIONS} -DHAVE_LONG_LONG")
+  endif (H5_SIZEOF_LONG_LONG)
+  TRY_RUN (HDF5_PRINTF_LL_TEST_RUN   HDF5_PRINTF_LL_TEST_COMPILE
+      ${HDF5_BINARY_DIR}/CMake
+      ${HDF5_RESOURCES_DIR}/HDF5Tests.c
+      CMAKE_FLAGS -DCOMPILE_DEFINITIONS:STRING=${CURRENT_TEST_DEFINITIONS}
+      OUTPUT_VARIABLE OUTPUT
+  )
+  if (HDF5_PRINTF_LL_TEST_COMPILE)
+    if (HDF5_PRINTF_LL_TEST_RUN MATCHES 0)
+      string(REGEX REPLACE ".*PRINTF_LL_WIDTH=\\[(.*)\\].*" "\\1" HDF5_PRINTF_LL "${OUTPUT}")
+      set (H5_PRINTF_LL_WIDTH "\"${HDF5_PRINTF_LL}\"" CACHE INTERNAL "Width for printf for type `long long' or `__int64', us. `ll")
+      set (PRINT_LL_FOUND 1)
+    else (HDF5_PRINTF_LL_TEST_RUN MATCHES 0)
+      message ("HDF5: Width test failed with result: ${HDF5_PRINTF_LL_TEST_RUN}")
+    endif (HDF5_PRINTF_LL_TEST_RUN MATCHES 0)
+  else (HDF5_PRINTF_LL_TEST_COMPILE)
+    file (APPEND ${CMAKE_BINARY_DIR}/CMakeFiles/CMakeError.log
+        "Test H5_PRINTF_LL_WIDTH failed with the following output:\n ${OUTPUT}\n"
     )
-    if (HDF5_PRINTF_LL_TEST_COMPILE)
-      if (HDF5_PRINTF_LL_TEST_RUN MATCHES 0)
-        set (H5_PRINTF_LL_WIDTH "\"${HDF5_PRINTF_LL}\"" CACHE INTERNAL "Width for printf for type `long long' or `__int64', us. `ll")
-        set (PRINT_LL_FOUND 1)
-      else (HDF5_PRINTF_LL_TEST_RUN MATCHES 0)
-        message ("Width with ${HDF5_PRINTF_LL} failed with result: ${HDF5_PRINTF_LL_TEST_RUN}")
-      endif (HDF5_PRINTF_LL_TEST_RUN MATCHES 0)
-    else (HDF5_PRINTF_LL_TEST_COMPILE)
-      file (APPEND ${CMAKE_BINARY_DIR}/CMakeFiles/CMakeError.log
-          "Test H5_PRINTF_LL_WIDTH for ${HDF5_PRINTF_LL} failed with the following output:\n ${OUTPUT}\n"
-      )
-    endif (HDF5_PRINTF_LL_TEST_COMPILE)
-  endforeach (HDF5_PRINTF_LL)
+  endif (HDF5_PRINTF_LL_TEST_COMPILE)
 
   if (PRINT_LL_FOUND)
-    message (STATUS "Checking for apropriate format for 64 bit long: found ${H5_PRINTF_LL_WIDTH}")
+    message (STATUS "Checking for appropriate format for 64 bit long: found ${H5_PRINTF_LL_WIDTH}")
   else (PRINT_LL_FOUND)
-    message (STATUS "Checking for apropriate format for 64 bit long: not found")
+    message (STATUS "Checking for appropriate format for 64 bit long: not found")
     set (H5_PRINTF_LL_WIDTH "\"unknown\"" CACHE INTERNAL
         "Width for printf for type `long long' or `__int64', us. `ll"
     )
   endif (PRINT_LL_FOUND)
 endif (NOT H5_PRINTF_LL_WIDTH OR H5_PRINTF_LL_WIDTH MATCHES "unknown")
diff -U5 -Nru hdf5-1.8.13/config/cmake/HDF5Tests.c hdf5-1.8.13.vs2015-support/config/cmake/HDF5Tests.c
--- hdf5-1.8.13/config/cmake/HDF5Tests.c	2014-05-05 22:13:20.000000000 -0400
+++ hdf5-1.8.13.vs2015-support/config/cmake/HDF5Tests.c	2016-08-08 15:43:40.332386621 -0400
@@ -232,11 +232,11 @@
 #include <stdlib.h>
 #include <string.h>
 
 int main(void)
 {
-  char *llwidthArgs[] = { "l64", "l", "L", "q", "ll", NULL };
+  char *llwidthArgs[] = { "I64", "l64", "l", "L", "q", "ll", NULL };
   char *s = malloc(128);
   char **currentArg = NULL;
   LL_TYPE x = (LL_TYPE)1048576 * (LL_TYPE)1048576;
   for (currentArg = llwidthArgs; *currentArg != NULL; currentArg++)
     {
