diff -u -U5 -r netcdf-c-4.7.0/include/nc4internal.h netcdf-c-4.7.0-fix-_win32-name/include/nc4internal.h
--- netcdf-c-4.7.0/include/nc4internal.h	2019-04-29 12:54:26.000000000 -0400
+++ netcdf-c-4.7.0-fix-_win32-name/include/nc4internal.h	2019-05-22 20:48:52.610037314 -0400
@@ -65,15 +65,15 @@
 #define X_LONG_MAX      X_INT_MAX
 #define X_UINT_MAX      4294967295U
 #define X_INT64_MIN     (-9223372036854775807LL-1LL)
 #define X_INT64_MAX     9223372036854775807LL
 #define X_UINT64_MAX    18446744073709551615ULL
-#ifdef WIN32 /* Windows, of course, has to be a *little* different. */
+#ifdef _WIN32 /* Windows, of course, has to be a *little* different. */
 #define X_FLOAT_MAX     3.402823466e+38f
 #else
 #define X_FLOAT_MAX     3.40282347e+38f
-#endif /* WIN32 */
+#endif /* _WIN32 */
 #define X_FLOAT_MIN     (-X_FLOAT_MAX)
 #define X_DOUBLE_MAX    1.7976931348623157e+308
 #define X_DOUBLE_MIN    (-X_DOUBLE_MAX)
 
 /** This is the number of netCDF atomic types. */
diff -u -U5 -r netcdf-c-4.7.0/libsrc/ncFile.c netcdf-c-4.7.0-fix-_win32-name/libsrc/ncFile.c
--- netcdf-c-4.7.0/libsrc/ncFile.c	2019-04-29 12:54:26.000000000 -0400
+++ netcdf-c-4.7.0-fix-_win32-name/libsrc/ncFile.c	2019-05-22 20:49:39.626320732 -0400
@@ -157,15 +157,15 @@
     if(state == NULL) return NC_EINVAL;
     if(state->file == NULL) return NC_EINVAL;
 #ifdef HAVE_FSYNC
 #ifdef USE_FSYNC
     fd = fileno(state->file);
-#ifndef WIN32
+#ifndef _WIN32
     fsync(fd);
 #else
     _commit(fd);
-#endif	/* WIN32 */
+#endif	/* _WIN32 */
 #endif
 #endif
     return NC_NOERR;
 }
 
diff -u -U5 -r netcdf-c-4.7.0/nc_test/tests.h netcdf-c-4.7.0-fix-_win32-name/nc_test/tests.h
--- netcdf-c-4.7.0/nc_test/tests.h	2019-04-29 12:54:26.000000000 -0400
+++ netcdf-c-4.7.0-fix-_win32-name/nc_test/tests.h	2019-05-22 20:52:04.079540132 -0400
@@ -41,15 +41,15 @@
 #define X_INT_MAX	2147483647
 #if defined(FLT_MAX_EXP) && FLT_MAX_EXP < 128
 /* FLT_MAX < X_FLOAT_MAX */
 #define X_FLOAT_MAX	FLT_MAX
 #else
-#ifdef WIN32 /* Windows, of course, has to be a *little* different. */
+#ifdef _WIN32 /* Windows, of course, has to be a *little* different. */
 #define X_FLOAT_MAX	3.402823466e+38f
 #else
 #define X_FLOAT_MAX	3.40282347e+38f
-#endif /* WIN32 */
+#endif /* _WIN32 */
 #endif
 #define X_FLOAT_MIN	(-X_FLOAT_MAX)
 #if CRAYFLOAT
 /* ldexp(1. - ldexp(.5 , -46), 1024) */
 #define X_DOUBLE_MAX    1.79769313486230e+308
