diff -NrU5 netcdf-4.3.2/config.h.in.cmake netcdf-4.3.2.VS2015/config.h.in.cmake
--- netcdf-4.3.2/config.h.in.cmake	2014-04-23 17:35:20.000000000 -0400
+++ netcdf-4.3.2.VS2015/config.h.in.cmake	2016-12-08 15:46:21.000000000 -0500
@@ -28,19 +28,23 @@
   #endif
   
 #ifdef _WIN32
 	#ifndef strcasecmp
   		#define strcasecmp _stricmp
-		#define snprintf _snprintf
+		#if defined(_MSC_VER) && (_MSC_VER < 1900)
+			#define snprintf _snprintf
+		#endif
   	#endif
 #endif
 
 
   #define strdup _strdup
   #define fdopen _fdopen
   #define write _write
-  #define snprintf _snprintf
+  #if defined(_MSC_VER) && (_MSC_VER < 1900)
+  	  #define snprintf _snprintf
+  #endif
   #define strtoll _strtoi64
 #endif
 
 
 /* Cache Size, other variables for HDF5 */
