diff -NrU5 qt-everywhere-opensource-src-4.8.4/src/3rdparty/clucene/src/CLucene/util/VoidMap.h qt-everywhere-opensource-src-4.8.4.VS2015/src/3rdparty/clucene/src/CLucene/util/VoidMap.h
--- qt-everywhere-opensource-src-4.8.4/src/3rdparty/clucene/src/CLucene/util/VoidMap.h	2012-11-23 05:10:12.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.4.VS2015/src/3rdparty/clucene/src/CLucene/util/VoidMap.h	2016-12-08 15:50:11.000000000 -0500
@@ -216,20 +216,20 @@
 #endif
 
 //A collection that contains no duplicates
 //does not guarantee that the order will remain constant over time
 template<typename _kt, typename _vt, 
-	typename _Compare,
-	typename _KeyDeletor=CL_NS(util)::Deletor::Dummy,
-	typename _ValueDeletor=CL_NS(util)::Deletor::Dummy>
+	typename x_Compare,
+	typename x_KeyDeletor=CL_NS(util)::Deletor::Dummy,
+	typename x_ValueDeletor=CL_NS(util)::Deletor::Dummy>
 class CLSet:public __CLMap<_kt,_vt,
-	CL_NS_STD(map)<_kt,_vt, _Compare>,
-	_KeyDeletor,_ValueDeletor>
+	CL_NS_STD(map)<_kt,_vt, x_Compare>,
+	x_KeyDeletor,x_ValueDeletor>
 {
-	typedef typename CL_NS_STD(map)<_kt,_vt,_Compare> _base;
-	typedef __CLMap<_kt, _vt, CL_NS_STD(map)<_kt,_vt, _Compare>,
-		_KeyDeletor,_ValueDeletor> _this;
+	typedef typename CL_NS_STD(map)<_kt,_vt,x_Compare> _base;
+	typedef __CLMap<_kt, _vt, CL_NS_STD(map)<_kt,_vt, x_Compare>,
+		x_KeyDeletor,x_ValueDeletor> _this;
 public:
 	CLSet ( const bool deleteKey=false, const bool deleteValue=false )
 	{
 		_this::setDeleteKey(deleteKey);
 		_this::setDeleteValue(deleteValue);
