diff -Nr -U5 szip-2.1/src/CMakeLists.txt szip-2.1.install-ricehdf5.h/src/CMakeLists.txt
--- szip-2.1/src/CMakeLists.txt	2011-05-17 14:35:35.000000000 -0400
+++ szip-2.1.install-ricehdf5.h/src/CMakeLists.txt	2015-12-02 12:13:28.844082843 -0500
@@ -39,10 +39,11 @@
 # Add file(s) to CMake Install 
 #-----------------------------------------------------------------------------
 INSTALL (
     FILES
         ${SZIP_PUBLIC_HEADERS}
+        ${SZIP_HDRS}
     DESTINATION
         ${SZIP_INSTALL_INCLUDE_DIR}
     COMPONENT
         headers
 )
