diff -u -Nru5 oneapi-tbb-2021.3.0-win/lib/cmake/tbb/TBBConfig.cmake oneapi-tbb-2021.3.0-win.cmake-relocate-win/lib/cmake/tbb/TBBConfig.cmake
--- oneapi-tbb-2021.3.0-win/lib/cmake/tbb/TBBConfig.cmake	2021-08-19 09:00:06.860470365 -0400
+++ oneapi-tbb-2021.3.0-win.cmake-relocate-win/lib/cmake/tbb/TBBConfig.cmake	2021-08-19 09:34:43.604301943 -0400
@@ -58,22 +58,24 @@
     set(_tbb_subdir intel64/${_tbb_subdir})
 else ()
     set(_tbb_subdir ia32/${_tbb_subdir})
 endif()
 
+set(_tbb_subdir "")
+
 foreach (_tbb_component ${TBB_FIND_COMPONENTS})
     set(TBB_${_tbb_component}_FOUND 0)
     
     unset(_bin_version)
     if (_tbb_component STREQUAL tbb)
         set(_bin_version ${_tbb_bin_version})
     endif()
 
-    get_filename_component(_tbb_release_lib "${_tbb_root}/redist/${_tbb_subdir}/${_tbb_component}${_bin_version}.dll" ABSOLUTE)
+    get_filename_component(_tbb_release_lib "${_tbb_root}/bin/${_tbb_subdir}/${_tbb_component}${_bin_version}.dll" ABSOLUTE)
 
     if (NOT TBB_FIND_RELEASE_ONLY)
-        get_filename_component(_tbb_debug_lib "${_tbb_root}/redist/${_tbb_subdir}/${_tbb_component}${_bin_version}_debug.dll" ABSOLUTE)
+        get_filename_component(_tbb_debug_lib "${_tbb_root}/bin/${_tbb_subdir}/${_tbb_component}${_bin_version}_debug.dll" ABSOLUTE)
     endif()
 
     if (EXISTS "${_tbb_release_lib}" OR EXISTS "${_tbb_debug_lib}")
         if (NOT TARGET TBB::${_tbb_component})
             add_library(TBB::${_tbb_component} SHARED IMPORTED)
