diff -u -Nr -U5 hdfsrc/CMakeFilters.cmake hdfsrc.szip-detection/CMakeFilters.cmake
--- hdfsrc/CMakeFilters.cmake	2023-08-15 09:02:05.000000000 -0400
+++ hdfsrc.szip-detection/CMakeFilters.cmake	2023-10-04 17:50:12.079919714 -0400
@@ -148,11 +148,17 @@
       endif ()
     endif ()
     if (SZIP_FOUND)
       set (SZIP_INCLUDE_DIR_GEN ${SZIP_INCLUDE_DIR})
       set (SZIP_INCLUDE_DIRS ${SZIP_INCLUDE_DIRS} ${SZIP_INCLUDE_DIR})
-      set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
+      if (SZIP_SHARED_LIBRARY)
+        set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_SHARED_LIBRARY})
+      elseif (SZIP_STATIC_LIBRARY)
+        set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_STATIC_LIBRARY})
+      else ()
+        set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
+      endif ()
     endif ()
   else ()
     if (BUILD_SZIP_WITH_FETCHCONTENT)
       # Only tgz files available
       ORIGINAL_SZIP_LIBRARY ("TGZ" ${HDF5_ENABLE_SZIP_ENCODING})
