diff -u -U5 -r hdf5-1.10.3/CMakeFilters.cmake hdf5-1.10.3.fix-ext-pkg-find/CMakeFilters.cmake
--- hdf5-1.10.3/CMakeFilters.cmake	2018-08-09 17:36:31.000000000 -0400
+++ hdf5-1.10.3.fix-ext-pkg-find/CMakeFilters.cmake	2019-03-15 11:53:03.238932850 -0400
@@ -39,13 +39,13 @@
 #-----------------------------------------------------------------------------
 option (HDF5_ENABLE_Z_LIB_SUPPORT "Enable Zlib Filters" ON)
 if (HDF5_ENABLE_Z_LIB_SUPPORT)
   if (NOT H5_ZLIB_HEADER)
     if (NOT ZLIB_USE_EXTERNAL)
-      find_package (ZLIB NAMES ${ZLIB_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
+      #find_package (ZLIB NAMES ${ZLIB_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
       if (NOT ZLIB_FOUND)
-        find_package (ZLIB) # Legacy find
+        find_package (ZLIB MODULE) # Legacy find
       endif ()
       if (ZLIB_FOUND)
         set (H5_ZLIB_HEADER "zlib.h")
         set (ZLIB_INCLUDE_DIR_GEN ${ZLIB_INCLUDE_DIR})
         set (ZLIB_INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS} ${ZLIB_INCLUDE_DIR})
@@ -97,13 +97,17 @@
     find_package (libaec 1.0.5 CONFIG)
     if (SZIP_FOUND)
       set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
     endif ()
     if (NOT SZIP_FOUND)
-      find_package (SZIP NAMES ${LIBAEC_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
+      #find_package (SZIP NAMES ${LIBAEC_PACKAGE_NAME}${HDF_PACKAGE_EXT} COMPONENTS static shared)
       if (NOT SZIP_FOUND)
-        find_package (SZIP) # Legacy find
+        if (WIN32)
+          find_package (SZIP CONFIG) # Legacy find
+        else ()
+          find_package (SZIP) # Legacy find
+        endif ()
       endif ()
     endif ()
     if (SZIP_FOUND)
       set (SZIP_INCLUDE_DIR_GEN ${SZIP_INCLUDE_DIR})
       set (SZIP_INCLUDE_DIRS ${SZIP_INCLUDE_DIRS} ${SZIP_INCLUDE_DIR})
