diff -u -Nr -U5 matplotlib-3.7.2/pyproject.toml matplotlib-3.7.2.no-certifi/pyproject.toml
--- matplotlib-3.7.2/pyproject.toml	2023-07-05 04:10:26.000000000 -0400
+++ matplotlib-3.7.2.no-certifi/pyproject.toml	2023-09-01 11:22:07.228721576 -0400
@@ -1,9 +1,9 @@
 [build-system]
 build-backend = "setuptools.build_meta"
 requires = [
-    "certifi>=2020.06.20",
+    #"certifi>=2020.06.20",
     "oldest-supported-numpy",
     "pybind11>=2.6",
     "setuptools>=42",
     "setuptools_scm>=7",
 ]
