diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f321a0..f79e44e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -958,17 +958,6 @@ OPTION(ENABLE_DAP_LONG_TESTS "Enable DAP long tests." OFF)
 OPTION(ENABLE_DAP_REMOTE_TESTS "Enable DAP remote tests." ON)
 SET(REMOTETESTSERVERS "remotetest.unidata.ucar.edu" CACHE STRING "test servers to use for remote test")
 
-# See if we have libzip
-FIND_PACKAGE(Zip)
-
-# Define a test flag for have curl library
-IF(Zip_FOUND)
-  INCLUDE_DIRECTORIES(${Zip_INCLUDE_DIRS})
-  SET(ENABLE_NCZARR_ZIP TRUE)
-ELSE()
-  SET(ENABLE_NCZARR_ZIP FALSE)
-ENDIF()
-
 # Enable some developer-only tests
 OPTION(ENABLE_EXTRA_TESTS "Enable Extra tests. Some may not work because of known issues. Developers only." OFF)
 IF(ENABLE_EXTRA_TESTS)
