diff -u -U5 -r netcdf-c-4.8.0/CMakeLists.txt src/CMakeLists.txt
--- netcdf-c-4.8.0/CMakeLists.txt	2021-03-30 16:23:10.000000000 -0400
+++ netcdf-c-4.8.0.hdf5-finding/CMakeLists.txt	2021-08-11 20:15:21.761543159 -0400
@@ -705,10 +705,14 @@
         ENDIF()
       ENDIF(${HDF5_VERSION} VERSION_GREATER "1.8.15")
 
     ELSE(MSVC)
 
+      IF(HDF5_hdf5_LIBRARY AND NOT HDF5_C_LIBRARY)
+        SET(HDF5_C_LIBRARY ${HDF5_hdf5_LIBRARY})
+      ENDIF()
+
       # Depending on the install, either HDF5_hdf_library or
       # HDF5_C_LIBRARIES may be defined.  We must check for either.
       IF(HDF5_C_LIBRARIES AND NOT HDF5_hdf5_LIBRARY)
         SET(HDF5_hdf5_LIBRARY ${HDF5_C_LIBRARIES})
       ENDIF()
@@ -725,11 +729,11 @@
   IF(NOT HDF5_C_LIBRARY_hdf5 OR "${HDF5_C_LIBRARY_hdf5}" STREQUAL "" )
     SET(HDF5_C_LIBRARY_hdf5 "${HDF5_C_LIBRARY}")
   ENDIF()
 
   #Check to see if H5Z_SZIP exists in HDF5_Libraries. If so, we must use szip library.
-  set (CMAKE_REQUIRED_INCLUDES ${HAVE_HDF5_H})
+  set (CMAKE_REQUIRED_INCLUDES ${HDF5_INCLUDE_DIR})
   CHECK_C_SOURCE_COMPILES("#include <H5public.h>
    #if !H5_HAVE_FILTER_SZIP
    #error
    #endif
    int main() {
