diff --git a/liblib/CMakeLists.txt b/liblib/CMakeLists.txt
index 6603bd0..fcd616a 100644
--- a/liblib/CMakeLists.txt
+++ b/liblib/CMakeLists.txt
@@ -57,6 +57,9 @@ IF(USE_HDF5 OR USE_NETCDF4)
     IF(HDF5_hdf5_hl_LIBRARY AND NOT HDF5_HL_LIBRARY)
       SET(HDF5_HL_LIBRARY ${HDF5_hdf5_hl_LIBRARY})
     ENDIF()
+    IF(HDF5_C_HL_LIBRARIES AND NOT HDF5_HL_LIBRARY)
+      SET(HDF5_HL_LIBRARY ${HDF5_C_HL_LIBRARIES})
+    ENDIF()
     SET(TLL_LIBS ${HDF5_C_LIBRARY} ${HDF5_HL_LIBRARY} ${TLL_LIBS} ${SZIP_LIBRARY})
   ELSE() # Windows CMake defines HDF5_LIBRARIES.
     SET(TLL_LIBS ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
