diff -Nr -U5 zlib-1.2.7/CMakeLists.txt zlib-1.2.7.version-script/CMakeLists.txt
--- zlib-1.2.7/CMakeLists.txt	2012-04-30 00:20:45.000000000 -0400
+++ zlib-1.2.7.version-script/CMakeLists.txt	2015-12-02 12:17:30.855308113 -0500
@@ -164,11 +164,14 @@
 endif()
 
 if(UNIX)
     # On unix-like platforms the library is almost always called libz
    set_target_properties(zlib zlibstatic PROPERTIES OUTPUT_NAME z)
-   set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/zlib.map")
+   if (NOT APPLE)
+     # FIXME: Don't add --version-script option on Macs.
+     set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,${CMAKE_CURRENT_SOURCE_DIR}/zlib.map")
+   endif()
 elseif(BUILD_SHARED_LIBS AND WIN32)
     # Creates zlib1.dll when building shared library version
     set_target_properties(zlib PROPERTIES SUFFIX "1.dll")
 endif()
 
