diff -Nr -U5 qt-everywhere-opensource-src-5.4.1/qtbase/src/plugins/platforms/cocoa/qcocoaeventdispatcher.mm qt-everywhere-opensource-src-5.4.1.osx-10.9-modal-dialog/qtbase/src/plugins/platforms/cocoa/qcocoaeventdispatcher.mm
--- qt-everywhere-opensource-src-5.4.1/qtbase/src/plugins/platforms/cocoa/qcocoaeventdispatcher.mm	2015-02-16 23:56:53.000000000 -0500
+++ qt-everywhere-opensource-src-5.4.1.osx-10.9-modal-dialog/qtbase/src/plugins/platforms/cocoa/qcocoaeventdispatcher.mm	2015-12-02 12:49:50.564237171 -0500
@@ -404,12 +404,38 @@
             // We can use exec-mode, meaning that we can stay in a tight loop until
             // interrupted. This is mostly an optimization, but it allow us to use
             // [NSApp run], which is the normal code path for cocoa applications.
             if (NSModalSession session = d->currentModalSession()) {
                 QBoolBlocker execGuard(d->currentExecIsNSAppRun, false);
-                while ([NSApp runModalSession:session] == NSModalResponseContinue && !d->interrupt)
-                    qt_mac_waitForMoreEvents(NSModalPanelRunLoopMode);
+
+                // OSX 10.9 appears to tighten up its internal code such that
+                // sessions shouldn't be used once they've been freed, which
+                // could happen in the else condition of the if(canExec)
+                // statement. Add extra logic to make sure we don't use the
+                // session once the current modal session is freed. We stop if
+                // the session is no longer current or if runModalSession()
+                // indicates the modal session is no longer continuing.
+                bool stayInLoop = true;
+                while (!d->interrupt && stayInLoop) {
+                    // Mavericks deprecated the status we wish to check.
+                    NSInteger checkStatus = 0;
+#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_8
+                    checkStatus = NSRunContinuesResponse;
+#else
+                    checkStatus = NSModalResponseContinue;
+#endif
+                    if([NSApp runModalSession:session] == checkStatus) {
+                        qt_mac_waitForMoreEvents();
+                    }
+                    else {
+                        stayInLoop = false;
+                    }
+
+                    if(session != d->currentModalSession()) {
+                        stayInLoop = false;
+                    }
+                }
 
                 if (!d->interrupt && session == d->currentModalSessionCached) {
                     // Someone called [NSApp stopModal:] from outside the event
                     // dispatcher (e.g to stop a native dialog). But that call wrongly stopped
                     // 'session' as well. As a result, we need to restart all internal sessions:
@@ -437,12 +463,20 @@
                 if (!excludeUserEvents) {
                     // Since we can dispatch all kinds of events, we choose
                     // to use cocoa's native way of running modal sessions:
                     if (flags & QEventLoop::WaitForMoreEvents)
                         qt_mac_waitForMoreEvents(NSModalPanelRunLoopMode);
+
+                    // Mavericks deprecated the status we wish to check.
+                    NSInteger checkStatus = 0;
+#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_8
+                    checkStatus = NSRunContinuesResponse;
+#else
+                    checkStatus = NSModalResponseContinue;
+#endif
                     NSInteger status = [NSApp runModalSession:session];
-                    if (status != NSModalResponseContinue && session == d->currentModalSessionCached) {
+                    if (status != checkStatus && session == d->currentModalSessionCached) {
                         // INVARIANT: Someone called [NSApp stopModal:] from outside the event
                         // dispatcher (e.g to stop a native dialog). But that call wrongly stopped
                         // 'session' as well. As a result, we need to restart all internal sessions:
                         d->temporarilyStopAllModalSessions();
                     }
