diff -U5 -Nru tbb44_20150728oss/include/tbb/tbb_config.h tbb44_20150728oss.warning-fix/include/tbb/tbb_config.h
--- tbb44_20150728oss/include/tbb/tbb_config.h	2015-08-24 04:14:49.000000000 -0400
+++ tbb44_20150728oss.warning-fix/include/tbb/tbb_config.h	2016-09-13 10:24:00.753258112 -0400
@@ -230,11 +230,11 @@
 #endif
 
 // C++11 standard library features
 
 #define __TBB_CPP11_VARIADIC_TUPLE_PRESENT          (!_MSC_VER || _MSC_VER >=1800)
-#define __TBB_CPP11_TYPE_PROPERTIES_PRESENT         (_LIBCPP_VERSION || _MSC_VER >= 1700)
+#define __TBB_CPP11_TYPE_PROPERTIES_PRESENT         (__GXX_EXPERIMENTAL_CXX0X__ && __TBB_GCC_VERSION >= 50100 || _LIBCPP_VERSION || _MSC_VER >= 1700)
 #define __TBB_TR1_TYPE_PROPERTIES_IN_STD_PRESENT    (__GXX_EXPERIMENTAL_CXX0X__ && __TBB_GCC_VERSION >= 40300 || _MSC_VER >= 1600)
 // GCC has a partial support of type properties
 #define __TBB_CPP11_IS_COPY_CONSTRUCTIBLE_PRESENT   (__GXX_EXPERIMENTAL_CXX0X__ && __TBB_GCC_VERSION >= 40700 || __TBB_CPP11_TYPE_PROPERTIES_PRESENT)
 
 // In GCC and MSVC, implementation of std::move_if_noexcept is not aligned with noexcept
