diff --git a/Makefile.in b/Makefile.in
index e11034b..2c6e54f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -5039,7 +5039,7 @@ PATHFILES = $(am__append_28) $(am__append_39) $(am__append_61) \
 ADOCFILES_COMMON = man-common/bugreports.adoc man-common/footer.adoc \
 	man-common/footer-config.adoc man-common/footer-lib.adoc \
 	man-common/manpage-stub.adoc man-common/translation.adoc
-MANPAGES = lib/terminal-colors.d.5 $(am__append_18) $(am__append_40) \
+MANPAGES = $(am__append_18) $(am__append_40) \
 	$(am__append_87) $(am__append_90) $(am__append_93) \
 	$(am__append_96) $(am__append_99) $(am__append_102) \
 	$(am__append_105) $(am__append_108) $(am__append_111) \
-- 
2.34.1
