diff --git a/Makefile.in b/Makefile.in
index e11034b..2c6e54f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -5316,7 +5316,7 @@ ADOCFILES_COMMON = man-common/bugreports.adoc man-common/colors.adoc \
 	man-common/footer-lib.adoc man-common/help-version.adoc \
 	man-common/in-bytes.adoc man-common/manpage-stub.adoc \
 	man-common/translation.adoc
-MANPAGES = lib/terminal-colors.d.5 $(am__append_20) $(am__append_44) \
+MANPAGES = $(am__append_20) $(am__append_44) \
 	$(am__append_91) $(am__append_94) $(am__append_97) \
 	$(am__append_100) $(am__append_103) $(am__append_106) \
 	$(am__append_109) $(am__append_112) $(am__append_115) \
-- 
2.34.1
