diff -u -Nr -U5 boost_1_83_0/tools/build/src/tools/darwin.jam boost_1_83_0.osx-headerpad_max_install_names/tools/build/src/tools/darwin.jam
--- boost_1_83_0/tools/build/src/tools/darwin.jam	2023-08-08 17:02:57.000000000 -0400
+++ boost_1_83_0.osx-headerpad_max_install_names/tools/build/src/tools/darwin.jam	2023-08-26 09:37:17.817618631 -0400
@@ -491,11 +491,11 @@
     prepare-framework-path $(<) ;
 }
 
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module -install_name "$(<:B)$(<:S)" -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
+    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module -headerpad_max_install_names -install_name "$(<:B)$(<:S)" -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
 }
 
 # We use libtool instead of ar to support universal binary linking
 # TODO: Find a way to use the underlying tools, i.e. lipo, to do this.
 actions piecemeal archive
diff -u -Nr -U5 boost_1_83_0/tools/build/src/tools/gcc.jam boost_1_83_0.osx-headerpad_max_install_names/tools/build/src/tools/gcc.jam
--- boost_1_83_0/tools/build/src/tools/gcc.jam	2023-08-08 17:02:57.000000000 -0400
+++ boost_1_83_0.osx-headerpad_max_install_names/tools/build/src/tools/gcc.jam	2024-05-02 17:14:50.821619296 -0400
@@ -872,11 +872,11 @@
     # at all and darwin.jam will use a separate 'strip' invocation.
     toolset.flags gcc.link RPATH <target-os>darwin : <dll-path> ;
     # This does not support -R.
     toolset.flags gcc.link RPATH_OPTION <target-os>darwin : -rpath ;
     # -rpath-link is not supported at all.
-    toolset.flags gcc.link SHARED_OPTION <target-os>darwin : -dynamiclib -single_module ;
+    toolset.flags gcc.link SHARED_OPTION <target-os>darwin : -Wl,-headerpad_max_install_names -dynamiclib -single_module ;
     toolset.flags gcc.link HAVE_SONAME <target-os>darwin : "" ;
     toolset.flags gcc.link SONAME_OPTION <target-os>darwin : -install_name ;
     toolset.flags gcc.link SONAME_PREFIX <target-os>darwin : @rpath/ ;
 
     # See note [1]
