diff -u -Nru5 freetype-2.10.2/configure freetype-2.10.2.64bit-inode/configure
--- freetype-2.10.2/configure	2020-03-22 15:24:22.000000000 -0400
+++ freetype-2.10.2.64bit-inode/configure	2021-04-07 20:52:42.349470882 -0400
@@ -91,11 +91,11 @@
 # build a dummy Makefile if we are not building in the source tree;
 # we use inodes to avoid issues with symbolic links
 inode_src=`ls -id $abs_ft2_dir | awk '{print $1}'`
 inode_dst=`ls -id $abs_curr_dir | awk '{print $1}'`
 
-if test $inode_src -ne $inode_dst; then
+if test $inode_src != $inode_dst; then
   if test ! -d docs; then
     mkdir docs
     echo "Copying documentation assets"
     cp -R $abs_ft2_dir/docs/markdown $abs_curr_dir/docs
   fi
