diff -u -Nru5 mpich-3.3/configure mpich-3.3.gfortran10/configure
--- mpich-3.3/configure	2018-11-21 12:38:39.000000000 -0500
+++ mpich-3.3.gfortran10/configure	2020-09-23 12:04:51.508915915 -0400
@@ -44399,11 +44399,11 @@
         # try again with -wmismatch=foo1
         save_FFLAGS="$FFLAGS"
 	# The best solution is to turn off errors on particular routines
 	# if that isn't possible (e.g., too many of them), then
 	# just try arguments that turn off all checking
-	for flags in  "-mismatch" ; do
+	for flags in  "-mismatch" "-fallow-argument-mismatch" ; do
             testok=no
             FFLAGS="$FFLAGS $flags"
             cat > conftest.$ac_ext <<_ACEOF
 
 
