diff -u -Nru5 Python-3.9.5/configure Python-3.9.5.extra-error-flag/configure
--- Python-3.9.5/configure	2021-05-03 10:54:42.000000000 -0400
+++ Python-3.9.5.extra-error-flag/configure	2021-07-20 21:15:48.522992570 -0400
@@ -7348,51 +7348,10 @@
     if test $ac_cv_enable_strict_prototypes_warning = yes
     then
       CFLAGS_NODIST="$CFLAGS_NODIST -Wstrict-prototypes"
     fi
 
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking if we can make implicit function declaration an error in $CC" >&5
-$as_echo_n "checking if we can make implicit function declaration an error in $CC... " >&6; }
-     ac_save_cc="$CC"
-     CC="$CC -Werror=implicit-function-declaration"
-     if ${ac_cv_enable_implicit_function_declaration_error+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-
-           ac_cv_enable_implicit_function_declaration_error=yes
-
-else
-
-           ac_cv_enable_implicit_function_declaration_error=no
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-fi
-
-     CC="$ac_save_cc"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_enable_implicit_function_declaration_error" >&5
-$as_echo "$ac_cv_enable_implicit_function_declaration_error" >&6; }
-
-    if test $ac_cv_enable_implicit_function_declaration_error = yes
-    then
-      CFLAGS_NODIST="$CFLAGS_NODIST -Werror=implicit-function-declaration"
-    fi
-
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking if we can use visibility in $CC" >&5
 $as_echo_n "checking if we can use visibility in $CC... " >&6; }
      ac_save_cc="$CC"
      CC="$CC -fvisibility=hidden"
      if ${ac_cv_enable_visibility+:} false; then :
